content/accessibility/truncation.mdx

```diff
@@ -0,0 +1,89 @@
+Ideally never truncate in the middle of a word, and even more ideally never truncate in the middle of a sentence or paragraph.
+
+For example, if a text includes mentions, truncation can happen at the middle of the mention, confusing the user who's reading the text.
+Or we truncate the name of a company, and we don't control where the truncation is happening, it might be that several companies with the similar beginning of their name have the same visual truncation, and that would be very confusing for the user.
+
+## Truncation options
+
+### CSS Truncation
+
+CSS provides mechanisms allowing you to visually hide or truncate content without needing to touch the underlying HTML.
+CSS truncation will not removed the truncated text from the DOM, so it could look unreadable for a sighted user, but screen reader users will still be able to read it.
+
+This method is purely controlled by the browser, and gives us no power on deciding where to truncate the text.
+
+We should not truncate large bodies of text using css. but there are times when we can truncate using CSS. If you're working with a user interface control where the truncated text is not controlled by designers and developers, for example a single line of text where cannot fit all the text on it, actually CSS truncation is the right thing to do because you're going to show the full thing in an accessible tooltip.
+
+If you're working with a user interface control where GitHub designers and developers have decided exactly what text it will have and exactly what its purpose is, meaning that is a generated content and it's known ahead of time, then we should avoid truncating it.
+
+#### How to truncate via CSS
+
+- `text-overflow: ellipsis` (combined with `overflow: hidden` and `white-space: nowrap`): This combination of attributes allows for easy visual truncation of long text which might overflow outside its container. However, since the text which doesn't fit never actually gets removed from the DOM, screen reader users will hear the entire thing when they navigate to it.
+
+Truncation can be achieve applying the [Primer CSS class for truncation](https://primer.style/css/components/truncate) to the text element, so when a text reaches lengths larger than existing container, it shortens with ellipses.
+
+### Progressive disclosure 
+
+[Primer Progressive Disclosure](https://primer.style/design/ui-patterns/progressive-disclosure) is an interaction design pattern that can be used for explicit truncation.
+
+### Using a tooltip
+
+When working with a user interface control where the truncated text is not controlled by designers and developers, for example focusable controls with user generated or unpredictable content in them, CSS truncation together with an accessible tooltip is a reasonable option.
+
+However, tooltips are rarely appropriate, and its misuse can result in a myriad of accessibility issues, so always consider not using a tooltip for an improved user experience.
```

Comment:
An example here to demonstrate the exact scenario where tooltip is appropriate would be helpful.